WebApplication/ClientApp/src/components/forgePdfView.js

```diff
@@ -80,32 +80,35 @@ export class ForgePdfView extends Component {
     componentDidUpdate(prevProps) {
         if (this.viewer && Autodesk && (this.props.drawingPdf !== prevProps.drawingPdf)) {
 
+            const findModelsForUrn = function(viewer, urn, onlyFirstModel) {
+                const allModels = viewer.getAllModels();
+                const models = [];
+                for (const model of allModels) {
+                    const modelUrn = model.getData().urn;
+                    if (modelUrn != urn)
+                        continue;
+
+                    models.push(model);
+
+                    if (onlyFirstModel === true)
+                        break;
+                }
+
+                return models;
+            };
+
             if (prevProps.drawingPdf != null) {
-                // try to find model in allModels
-                const allModels = this.viewer.getAllModels();
-                allModels.forEach(model => {
-                    const urn = model.getData().urn;
-                    if (urn != prevProps.drawingPdf)
-                        return;
-
-                    this.viewer.hideModel(model);
-                });
+                // try to find model(s) in allModels and hide all of them
```

Comment:
Could you clarify: it means that viewer can contains several visible models with the same URN?

Reply:
yes, I think that it was in case that were opened multiple sheets from one source (I saw it sometimes, sometimes it is only one)
- after unload it is somehow managed, I found that sometimes after hide, I don't have all I opened (maybe it depends on complexity and size of pdf)
- for show it is enough to do it on the first one I found

Reply:
David, could you please put this explanation into comment in the source code?  I think it's important to not forget and accidentally "optimize" the code :)

Reply:
I was playing with more multisheet drawings and unable to get this more models state, so maybe it was in case that I did hide/show wrong way, so I will change it to get always the first one only.